iroha_client/tests/tests/tests.rs

```diff
@@ -13,3 +29,271 @@ mod transfer_asset;
+        let mut configuration = ClientConfiguration::test(api_url);
+        configuration.account_id = account_id.clone();
+        configuration.public_key = keys.public_key;
+        configuration.private_key = keys.private_key;
+        Client::new(&configuration)
+    }
+
+    fn loop_on_events(mut self, event_filter: EventFilter, f: impl Fn(Result<Event>)) {
+        for event in self
+            .listen_for_events(event_filter)
+            .expect("Failed to create event iterator")
+        {
+            f(event)
+        }
+    }
+
+    fn submit_till(
+        &mut self,
+        instruction: impl Into<Instruction> + Debug,
+        request: &QueryRequest,
+        f: impl Fn(&QueryResult) -> bool,
+    ) -> QueryResult {
+        self.submit(instruction)
+            .expect("Failed to submit instruction");
+        self.poll_request(request, f)
+    }
+
+    fn submit_all_till(
+        &mut self,
+        instructions: Vec<Instruction>,
+        request: &QueryRequest,
+        f: impl Fn(&QueryResult) -> bool,
+    ) -> QueryResult {
+        self.submit_all(instructions)
+            .expect("Failed to submit instruction");
+        self.poll_request(request, f)
+    }
+
+    fn poll_request_with_multiplier(
+        &mut self,
+        request: &QueryRequest,
+        multiplier: u32,
+        f: impl Fn(&QueryResult) -> bool,
+    ) -> QueryResult {
+        let tries = 10;
+        let pipeline_time = Configuration::pipeline_time() * multiplier / tries;
+        for _ in 0..tries {
+            thread::sleep(pipeline_time);
+            if let Ok(result) = self.request(request) {
+                if f(&result) {
+                    return result;
+                }
+            }
+        }
+
+        panic!("Failed to wait till consensus")
+    }
+
+    fn poll_request(
+        &mut self,
+        request: &QueryRequest,
+        f: impl Fn(&QueryResult) -> bool,
+    ) -> QueryResult {
+        self.poll_request_with_multiplier(request, 2, f)
+    }
+}
+
+impl TestQueryResult for QueryResult {
+    fn find_asset_by_id(&self, asset_id: &AssetDefinitionId) -> Option<&Asset> {
+        let assets = if let QueryResult(Value::Vec(assets)) = self {
+            assets
+        } else {
+            panic!("Wrong Query Result Type.");
+        };
+        assets.iter().find_map(|asset| {
+            if let Value::Identifiable(IdentifiableBox::Asset(ref asset)) = asset {
+                if &asset.id.definition_id == asset_id {
+                    return Some(asset.as_ref());
+                }
+            }
+            None
+        })
+    }
+}
+
+#[allow(clippy::clippy::cast_possible_truncation)]
+impl TestNetwork for Network {
+    fn start_test(n_peers: usize, maximum_transactions_in_block: u32) -> (Self, Client) {
+        let mut configuration = Configuration::test();
+        // Reverse consensus
+        configuration.sumeragi_configuration.max_faulty_peers = ((n_peers - 1) / 3) as u32;
```

Comment:
What do you mean by `Reverse consensus`?

Reply:
I mean that we can calculate how many peers can be faulty from consensus requirements

Reply:
Better to write it then like `Calculates max faulty peers based on consensus requirements`.